test/test-parse-statement.cpp

```diff
@@ -155,17 +155,28 @@ TEST(test_parse, return_statement_disallows_newline) {
            u8"typeof banana",
            u8"{}",
            u8"~bits",
+           u8"<div>hi</div>",
+           u8"<a href=\"github.com/quick-lint/quick-lint-js\">link</a>",
+           u8"<p></p>",
+           u8"<div>this is <b>bold</b> and <i>italic</i> text</div>",
+           u8"<head><title>head and title</title></head>",
            // TODO(strager): Contextual keywords (let, from, yield, etc.).
            // TODO(strager): Function without name. (Must be an expression, not
            // a statement.)
-           // TODO(strager): JSX.
        }) {
     {
       padded_string code(u8"return\n"s + second_line);
       SCOPED_TRACE(code);
       spy_visitor v;
-      parser p(&code, &v);
-      p.parse_and_visit_module(v);
+      if (second_line[0] == '<') {
+        parser_options options;
+        options.jsx = true;
+        parser p(&code, &v, options);
```

Comment:
Minor: We can use the [`jsx_options` variable](https://github.com/quick-lint/quick-lint-js/blob/86cbabefe61baf2f6dd164f36cf362fa1dce76b2/test/quick-lint-js/parse-support.h#L119):
```suggestion
        parser p(&code, &v, jsx_options);
```